honeybadgermpc/batch_reconstruction.py

```diff
@@ -126,6 +128,106 @@ def with_at_most_non_none(data, k):
             yield x
 
 
+def optimistic_reconstruct(data, field, point, n, t):
+    """Reconstruct polynomial from t+1 values
+
+    optimistic_reconstruct interpolates a polynomial from t+1 values i.e. it assumes
+    that no malicious / erroneous data is present in the first t+1 values for each chunk.
+    It is the caller's responsibility to verify if the polynomial is correct
+
+    :param data: Data received from parties
+        data[i] = data received from party i
+    :type field: honeybadgermpc.field.Field
+    :param point: Evaluation point information
+    :type point: honeybadgermpc.polynomial.EvalPoint
+    :param n: Number of parties
+    :param t: Threshold / Maximum number of malicious parties
+    :return: polynomials, evaluations
+    """
+    n_chunks = max(len(d) for d in data if d is not None)
+    x = []
+    z = []
+    chunks = [[] for _ in range(n_chunks)]
+    for i, d in enumerate(data):
+        if d is None:
+            continue
+
+        assert len(d) == n_chunks
+        x.append(point(i).value)
+        z.append(i)
+        for j in range(n_chunks):
+            chunks[j].append(d[j])
+        if len(x) == t + 1:
+            break
+
+    assert len(x) == t + 1
+
+    if point.use_fft:
+        # FFT
+        p = field.modulus
+        polynomial_solutions = fft_batch_interpolate(z, chunks, point.omega.value,
+                                                     p, point.order)
+        evaluations = [fft(coeffs, point.omega.value, p, point.order)[:n]
+                       for coeffs in polynomial_solutions]
+    else:
+        # Non-FFT
+        polynomial_solutions = batch_vandermonde_interpolate(x, chunks, field.modulus)
+        evaluations = batch_vandermonde_evaluate([point(i).value for i in range(n)],
+                                                 polynomial_solutions, field.modulus)
+
+    return polynomial_solutions, evaluations
+
+
+def merge_lists(lists):
+    result = []
+    for l in lists:
+        result += l
+    return result
+
+
+async def batch_interpolate(data_receivers, field, point, n, t):
```

Comment:
Thanks --- this is a much improved refactor already just in the python code. Having it previous part of the "wb_interpolate" world was the wrong choice, so this is better